Compiling on Fedora 40 fails

# Environment
- Perl 5.38.2
- Fedora 40
- GCC 14.2.1

- DBI master at 16a64d022138be14d40ce4d009c3268ac25d7797 (also happens with 1.644 tag)

# Failure

```
In file included from /usr/lib64/perl5/CORE/perl.h:6227:
DBI.xs: In function ‘dbih_inner’:
/usr/lib64/perl5/CORE/hv.h:541:31: error: passing argument 2 of ‘Perl_hv_common_key_len’ from incompatible pointer type [-Wincompatible-pointer-types]
  541 |     ((SV**) hv_common_key_len((hv), (key), (klen), (lval)               \
      |                               ^~~~
      |                               |
      |                               SV * {aka struct sv *}
/usr/lib64/perl5/CORE/embed.h:293:78: note: in definition of macro ‘hv_common_key_len’
  293 | # define hv_common_key_len(a,b,c,d,e,f)         Perl_hv_common_key_len(aTHX_ a,b,c,d,e,f)
      |                                                                              ^
DBI.xs:1109:14: note: in expansion of macro ‘hv_fetch’
 1109 |         if (!hv_fetch(ohv,"_NO_DESTRUCT_WARN",17,0))
```

--- a/DBI.xs
+++ b/DBI.xs
@@ -1106,7 +1106,7 @@ dbih_inner(pTHX_ SV *orv, const char *what)
     if (!SvMAGICAL(ohv)) {
         if (!what)
             return NULL;
-        if (!hv_fetch(ohv,"_NO_DESTRUCT_WARN",17,0))
+        if (!hv_fetch((HV*)ohv,"_NO_DESTRUCT_WARN",17,0))
 	    sv_dump(orv);
         croak("%s handle %s is not a DBI handle (has no magic)",
                 what, neatsvpv(orv,0));

